### PR TITLE
Add observer integration tests

### PR DESCRIPTION
**Pull request by miklosszegedi**
This is just the framework for the actual observer tests. It runs access tests but it shows the final pattern and test set.

### CODE REVIEW

File: module/mock/hot_stuff_follower.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Code generated by mockery v1.0.0. DO NOT EDIT.
```
Comment (koko1123) [inline module/mock/hot_stuff_follower.go:1]: why are we regenerating this?

File: integration/go.mod [hunk 2]
```diff
@@ -25,7 +25,7 @@ require (
 	github.com/onflow/flow/protobuf/go/flow v0.3.1
 	github.com/plus3it/gorecurcopy v0.0.1
 	github.com/prometheus/client_golang v1.12.1
-	github.com/rs/zerolog v1.26.1
```
Comment (koko1123) [inline integration/go.mod:28]: not sure why you're upgrading `zerolog`?

File: integration/tests/observer/execution_state_sync_test.go [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+		ed, err := eds.Get(ctx, result.ExecutionDataID)
+		assert.NoError(s.T(), err, "could not get execution data for height %v", i)
+
+		s.T().Logf("got execution data for height %d", i)
+		assert.Equal(s.T(), header.ID(), ed.BlockID)
+	}
+}
+
+func (s *ExecutionStateSyncSuite) nodeExecutionDataService(node *testnet.Container) (state_synchronization.ExecutionDataService, irrecoverable.SignalerContext) {
```
Comment (koko1123) [inline integration/tests/observer/execution_state_sync_test.go:177]: rename this to something like `getTestExecutionDataServiceNode`

File: module/mock/hot_stuff_follower.go (hunk 1 above)
Comment (szegedim) [reply module/mock/hot_stuff_follower.go:1]: I think I just wanted to make sure that all Makefile options still work cloning the observer tests from access. I can remove it from the change.

File: integration/go.mod (hunk 2 above)
Comment (szegedim) [reply integration/go.mod:28]: I think I just wanted to make sure that all Makefile options still work cloning the observer tests from access. I remember I made a go mod tidy. I can remove it from the change.

Comment (szegedim) [reply integration/go.mod:28]: However, it is likely that future changes that include tidy results would do the same.

File: module/mock/hot_stuff_follower.go (hunk 1 above)
Comment (szegedim) [reply module/mock/hot_stuff_follower.go:1]: Fixed in: https://github.com/GetElastech/flow-go/commit/5078c8c76b61ec4fb9acf3caefecb365c358b1ba

File: integration/go.mod (hunk 2 above)
Comment (szegedim) [reply integration/go.mod:28]: Fixed in: https://github.com/GetElastech/flow-go/commit/5078c8c76b61ec4fb9acf3caefecb365c358b1ba

File: integration/tests/observer/execution_state_sync_test.go (hunk 3 above)
Comment (szegedim) [reply integration/tests/observer/execution_state_sync_test.go:177]: Fixed in: https://github.com/GetElastech/flow-go/commit/5078c8c76b61ec4fb9acf3caefecb365c358b1ba

**Comment by miklosszegedi** [comment]
This change is superseded by https://github.com/onflow/flow-go/pull/2901

**Pull request closed without merging**